Benchmarks/TypeEqualityChecking/TypeEqualityCheckingBenchmarks.cs

```diff
@@ -0,0 +1,130 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//	This code was generated from the "TypeEqualityCheckingBenchmarks.tt" T4 Text Template.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+using BenchmarkDotNet.Attributes;
+using System.Runtime.CompilerServices;
+using System;
+
+namespace TypeEqualityChecking
+{
+	public class TypeEqualityCheckingBenchmarks
+	{
+		bool temp;
+
+		[Benchmark]
+		public void TypeOfEquals()
+		{
+			// true checks
+			temp = typeof(string) == typeof(string);
+			temp = typeof(int) == typeof(int);
+			temp = typeof(double) == typeof(double);
+			temp = typeof(decimal) == typeof(decimal);
+			temp = typeof(byte) == typeof(byte);
+			temp = typeof(short) == typeof(short);
+
+			// false checks
+			temp = typeof(string) == typeof(int);
+			temp = typeof(int) == typeof(string);
+			temp = typeof(double) == typeof(int);
+			temp = typeof(decimal) == typeof(int);
+			temp = typeof(byte) == typeof(int);
+			temp = typeof(short) == typeof(int);
+		}
+
+		static bool TypeOfEqualsGeneric<A, B>() => typeof(A) == typeof(B);
+
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		static bool TypeOfEqualsGenericInlined<A, B>() => typeof(A) == typeof(B);
+
+		[Benchmark]
+		public void TypeOfEqualsGeneric()
+		{
+			// true checks
+			temp = TypeOfEqualsGeneric<string, string>();
+			temp = TypeOfEqualsGeneric<int, int>();
+			temp = TypeOfEqualsGeneric<double, double>();
+			temp = TypeOfEqualsGeneric<decimal, decimal>();
+			temp = TypeOfEqualsGeneric<byte, byte>();
+			temp = TypeOfEqualsGeneric<short, short>();
+
+			// false checks
+			temp = TypeOfEqualsGeneric<string, int>();
+			temp = TypeOfEqualsGeneric<int, string>();
+			temp = TypeOfEqualsGeneric<double, int>();
+			temp = TypeOfEqualsGeneric<decimal, int>();
+			temp = TypeOfEqualsGeneric<byte, int>();
+			temp = TypeOfEqualsGeneric<short, int>();
+		}
+
+		[Benchmark]
+		public void TypeOfEqualsGenericInlined()
+		{
+			// true checks
+			temp = TypeOfEqualsGenericInlined<string, string>();
+			temp = TypeOfEqualsGenericInlined<int, int>();
+			temp = TypeOfEqualsGenericInlined<double, double>();
+			temp = TypeOfEqualsGenericInlined<decimal, decimal>();
+			temp = TypeOfEqualsGenericInlined<byte, byte>();
+			temp = TypeOfEqualsGenericInlined<short, short>();
+
+			// false checks
+			temp = TypeOfEqualsGenericInlined<string, int>();
+			temp = TypeOfEqualsGenericInlined<int, string>();
+			temp = TypeOfEqualsGenericInlined<double, int>();
+			temp = TypeOfEqualsGenericInlined<decimal, int>();
+			temp = TypeOfEqualsGenericInlined<byte, int>();
+			temp = TypeOfEqualsGenericInlined<short, int>();
+		}
+
+		struct Operator_is_struct<A, B> { }
+		static bool Operator_is<A, B>() => default(Operator_is_struct<A, B>) is Operator_is_struct<A, A>;
```

Comment:
I don't think we need it anymore, to be honest. I'd remove it

Reply:
It depends on what the goal of the benchmark is. Personally I think it should stay on the topic of type equality checking. If you think we should make the benchmark represent something else (such as examples where the JIT can't inline) then that would IMO require a lot more examples.